contracts/modules/LoanClosingsWith.sol

```diff
@@ -104,14 +106,20 @@ contract LoanClosingsWith is LoanClosingsShared {
         )
     {
         _checkAuthorized(loanId);
-        return
+        CloseWithSwapReturn memory closeWithSwapReturn =
```

Comment:
why did you introduce the CloseWithSwapReturn struct? Is it because of the invariant check?
I think it does not help readability. By moving the invariant checks out of the actual closing function you're solving the stack too deep problem in a nicer way.

Reply:
There was stack too deep issue, not even 1 local variable can be added.

Reply:
the additional gas cost shouldn't be too high and the improved readability and reusability is worth it, i think.

Reply:
Updated